lib/foreman/http_proxy.rb

```diff
@@ -2,7 +2,7 @@ module Foreman
   module HTTPProxy
     def http_proxy
       ActiveRecord::Base.connection_pool.with_connection do
-        Setting[:http_proxy]
+        Setting[:http_proxy].presence
```

Comment:
this now returns true/false but [later](https://github.com/theforeman/foreman/pull/6900/files#diff-c4a0e949d8924bdb7fce6a22cb14a2a0R17) we do `!http_proxy.nil?`

I think we either need to return true/nil here or modify `proxy_http_request?` method too

Reply:
That would be in case we would use `present?`. But (`presence`)[https://apidock.com/rails/Object/presence] is actually implemented as `self if self.present?` so it will result in `nil` if `presence?` would be falsy, and original value otherwise. And I guess we want exactly that.

Reply:
oh I misread that, sorry, you're absolutety right, I'll give it a test and will proceed with the merge if that works (chances are it will :-)